File: AOJ/ITP1/Transformation.cpp
#include<iostream>
#include <string>
using namespace std;

int main(void){
    string str;
    int n;
    string command;
    int arg1,arg2;
    string replace;
    cin>>str;
    cin>>n;
    for(int i=0;i<n;i++){
        cin>>command>>arg1>>arg2;
        if(command=="replace"){
                cin>>replace;
                str=str.substr(0,arg1)+replace+str.substr(arg2+1,-1);
            }
        else if(command=="reverse"){
replace=str.substr(arg1,arg2+1);
                reverse(replace.begin(),replace.end());
                str=str.substr(0,arg1)+replace+str.substr(arg2+1,-1);
            }
        else if(command=="print"){
                cout<<str.substr(arg1,arg2+1)<<endl;
        }
    }
    return 0;
}

File: AOJ/ITP1/Transformation.py
s=input()
q=int(input())
for i in range(q):
    t=input().split(" ")
    if t[0]=="print":
        a,b=int(t[1]),int(t[2])
        print(s[a:b+1:])
    if t[0]=="reverse":
        a,b=int(t[1]),int(t[2])
        s=s[:a:]+s[a:b+1:][::-1]+s[b+1::]
    if t[0]=="replace":
        a,b=int(t[1]),int(t[2])
        x=t[3]
        s=s[:a:]+x+s[b+1::]
